File: JS/Dragao.test.js
/**
 * @version JAVASCRIPT
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
const TEST = require('tape')
const OBJ = require('./Dragao.js')

TEST('Starting Dragao test...', (t) => {
    //Test 1
    t.assert(OBJ.checkEnergyLevel(8001) === "Mais de 8000!", "Expect: 8001 = Mais de 8000!")
    //Test 2
    t.assert(OBJ.checkEnergyLevel(100) === "Inseto!", "Expect: 100 = Inseto!")
    //Test 3
    t.assert(OBJ.checkEnergyLevel(200) === "Inseto!", "Expect: 200 = Inseto!")

    t.end()
})

File: GO/Dragao_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import "testing"

func TestDragao(t *testing.T) {
	tables := []struct {
		energy int
		want   string
	}{
		{8001, "Mais de 8000!"},
		{100, "Inseto!"},
		{200, "Inseto!"},
	}

	for _, table := range tables {
		got := checkEnergyLevel(table.energy)

		if got != table.want {
			t.Errorf("Waiting for this %s but the return was this: %s", table.want, got)
		}
	}
}
